examples/stress_tests/many_lights.rs

```diff
@@ -170,8 +170,8 @@ fn print_visible_light_count(
     }
 }
 
-fn extract_time(mut commands: Commands, time: Res<Time>) {
-    commands.insert_resource(time.into_inner().clone());
+fn extract_time(mut commands: Commands, time: Extract<Res<Time>>) {
+    commands.insert_resource(time.clone());
```

Comment:
Not a huge fan of just re-using the time resource across worlds - we don't own the Time resource to define what it means in the context of the render world

This is only a stress test, so it's probably fine. But we should at least have a comment.

Reply:
I disagree. In my mind, the time used for simulation at the point that a frame is to be rendered (worded that way because maybe simulation runs faster than rendering as it has in some games where they wanted 200Hz physics and 30/60 Hz rendering) is the same time used for rendering. We will likely have shaders that depend on time / frame number or so at some point and I expect that we should extract these by default into the render world from the main world. If people then want to do 'unusual' things with time like Max Payne bullet-time stuff, but do it by modifying the rate or even direction of the world's time progression, then maybe they'd need to consider more whether they should use the same time for rendering or not.

Reply:
I disagree. In my mind, the time used for simulation at the point that a frame is to be rendered (worded that way because maybe simulation runs faster than rendering as it has in some games where they wanted 200Hz physics and 30/60 Hz rendering) is the same time used for rendering. We will likely have shaders that depend on time / frame number or so at some point and I expect that we should extract these by default into the render world from the main world. If people then want to do 'unusual' things with time like Max Payne bullet-time stuff, but do it by modifying the rate or even direction of the world's time progression, then maybe they'd need to consider more whether they should use the same time for rendering or not.

Reply:
Whether we should extract the `Time` resource as-is is a different question to whether we should extract the Time resource only *in an example*.

I don't disagree that the time resource could reasonably be extracted into the time resource. However, I am claiming that the example legislating that unilaterally is not good

Reply:
I feel like I’m not understanding _why_ you think it’s not good. Is it because by putting it in an example you think we’re recommending that people add such an extract system in their own apps and you think it should rather be something that is there in the core of bevy so that we ensure all the timing code is correct? Or what is the reason?

Also, did my previous comment really send 3 times? That’s very weird. Apologies for that.

Reply:
Code outside of `bevy_time` (or `bevy_render` maybe?) *cannot* decide what the meaning of the `Time` resource is when inserted in the render world - it isn't defined.
Multiple crates could disagree on the semantics, if we allowed this. Say crate A extracts `Time` each frame, but crate B uses it to store the `Time` at the time idk, the current level was entered.
(Obviously in this case, crate a has the 'correct' semantics).
My claim is that other cases would not be so clear cut, and it's irresponsible to include this here in an example without such a discussion.

Now potentially, `bevy_render` could decree that the value of resources in the render world *must* be the same as the value in the simulation world at extraction time. That *could* be fine, and make this pattern allowed.
I don't however believe that we do claim this. At the very least, it's not documented on [RenderStage::Extract](http://dev-docs.bevyengine.org/bevy/render/enum.RenderStage.html#variant.Extract).

(If it were at all feasible, I'd love to make '`RenderWorldResource`' be a trait, so that you had to explicitly define that it can be added to the render world. We should still follow the orphan rules in lieu of this being statically checked)

Reply:
Thanks, that's much clearer. ❤️

Reply:
Reasonable take. I'll add a wrapper struct for now. Ideally in the near future we'll have renderer-side time standardized (including time uniforms for shaders). Seems like super low hanging fruit that makes a bunch of common scenarios much easier.